app/models/vm_scan.rb

```diff
@@ -102,6 +102,8 @@ def call_snapshot_create
       if vm.kind_of?(ManageIQ::Providers::Openstack::CloudManager::Vm) ||
          vm.kind_of?(ManageIQ::Providers::Microsoft::InfraManager::Vm)
         return unless create_snapshot(vm)
+      elsif vm.kind_of?(ManageIQ::Providers::Azure::CloudManager::Vm) && vm.require_snapshot_for_scan?
```

Comment:
why not just have each type of `Vm` have a `require_snapshot_for_scan` method and remove all this complex if/else/elsif logic?

Reply:
Ideally, we'd have different scan job classes based on VM type. The jobs for the types of VMs that don't require snapshots won't even have a snapshot state.